sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -829,13 +833,35 @@ private Query getQuery(Function inner, Context context) {
                             fieldData
                     );
                 } else {
+                    final Version indexCreated = Version.indexCreated(context.indexCache.indexSettings());
                     Coordinate[] coordinates = geometry.getCoordinates();
-                    GeoPoint[] points = new GeoPoint[coordinates.length];
-                    for (int i = 0; i < coordinates.length; i++) {
-                        Coordinate coordinate = coordinates[i];
-                        points[i] = new GeoPoint(coordinate.y, coordinate.x);
+                    final Query query;
+                    if (indexCreated.before(Version.V_2_2_0)) {
+                        GeoPoint[] points = new GeoPoint[coordinates.length];
+                        for (int i = 0; i < coordinates.length; i++) {
```

Comment:
can you extract some of the logic here into separate method? getQuery is already rather large.